modules/account-lib/src/coin/sol/instructionParser.ts

```diff
@@ -0,0 +1,79 @@
+import { SystemInstruction, TransactionInstruction } from '@solana/web3.js';
+
+import { TransactionType } from '../baseCoin';
+import { NotSupported } from '../baseCoin/errors';
+import { InstructionBuilderTypes, ValidInstructionTypesEnum } from './constants';
+import { InstructionParams, WalletInit, Transfer, Nonce, Memo } from './iface';
+import { getInstructionType } from './utils';
+
+export function instructionParser(type: TransactionType, instructions: TransactionInstruction[]): InstructionParams[] {
+  const instructionData: InstructionParams[] = [];
+  if (type === TransactionType.WalletInitialization) {
+    const walletInit = parseWalletInitTransaction(instructions);
+    instructionData.push(walletInit);
```

Comment:
Here we are have two responsabilities: one for parsing and other to pushing into instructionData. I think this should be on separate methods.

Reply:
Renamed